pkg/cli/build.go

```diff
@@ -108,14 +115,23 @@ func addDockerfileFlag(cmd *cobra.Command) {
 
 func addUseCogBaseImageFlag(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&buildUseCogBaseImage, "use-cog-base-image", false, "Use pre-built Cog base image for faster cold boots")
-	cmd.Flags().VisitAll(func(f *pflag.Flag) {
-		if f.Name == "use-cog-base-image" {
-			f.Hidden = true
-		}
-	})
 }
 
 func addBuildTimestampFlag(cmd *cobra.Command) {
 	cmd.Flags().Int64Var(&config.BuildSourceEpochTimestamp, "timestamp", -1, "Number of seconds sing Epoch to use for the build timestamp; this rewrites the timestamp of each layer. Useful for reproducibility. (`-1` to disable timestamp rewrites)")
 	_ = cmd.Flags().MarkHidden("timestamp")
 }
+
+func checkMutuallyExclusiveFlags(cmd *cobra.Command) error {
+	flags := []string{"use-cog-base-image", "use-cuda-base-image", "dockerfile"}
```

Comment:
Can these be programmatically determined? I'm ok with them being hardcoded for a first pass as we go, but it might be helpful to break this out into separate logic

Reply:
Can you give an example of what you are thinking in terms of improvement?

Reply:
Not sure if that would add much here as those commands are only part of this package and relationship between the flags (what makes them mutually exclusive) is very specific to the code here.